cmd/tnf/claim/compare/nodes/cnis/cnis.go

```diff
@@ -0,0 +1,311 @@
+package cnis
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"sort"
+
+	"github.com/test-network-function/cnf-certification-test/cmd/tnf/pkg/claim"
+)
+
+const (
+	elemNotFoundIn = "not found in "
+)
+
+// CNI Network config sections
+const (
+	differentCNIVersion   = "cniVersion"
+	differentPlugins      = "plugins"
+	differentDisableCheck = "disable_check"
+)
+
+// CNINetworkDiffReport holds the differences report for a cni network.
+// This is the structure of a CNI network config:
+// - name
+// - cniVersion
+// - disableCheck
+// - plugins
+// See https://github.com/containernetworking/cni/blob/main/SPEC.md#section-1-network-configuration-format
+// The Differences slice shows which of those fields have a different value
+// in the claim files for that network of the same node.
+type CNINetworkDiffReport struct {
+	NetworkName        string                `json:"networkName"`
+	Differences        []string              `json:"differences"`
+	PluginsDiffReports CNIPluginsDiffReports `json:"pluginsDiffReport,omitempty"`
+}
+
+// CNIPluginDiffReport holds the differences report for a cni network plugin.
+// Plugins' config have some known fields (described in the spec) but can also
+// have custom fields whose attributes' names and values' types are not known.
+// Only the plugin itself knows the meaning of those fields.
+// The Differences slice will hold the name of each attribute whoe value is
+// different.
+type CNIPluginDiffReport struct {
+	PluginName  string   `json:"pluginName"`
+	Differences []string `json:"differences"`
+}
+
+type CNINetworksDiffReports []CNINetworkDiffReport
+type CNIPluginsDiffReports []CNIPluginDiffReport
+
+// Stringer method to show in a table the fields where cni networks are different.
+// The output looks like this:
+//
+// CNI-NETWORK                   DIFFERENCES
+// crio                          cniVersion,plugins
+// loopback						 not found in claim2
+//
+// That example shows that the crio network is not the same in both claim files
+// for a given node. The difference column shows a list of "fields" whose value
+// is different. Plugins means that there's at least one crio plugin whose config
+// is different.
+// In case the network appears only in one claim file, the only difference will
+// be "not found in claim[1|2]"
+func (networksDiff CNINetworksDiffReports) String() string {
```

Comment:
I feel like  the table format output is missing details to be clear enough. Why not print the json path of the exact object that has differences at every level for instance at plugin level (there might be a better/simple jq command) :
'.claim.nodes.cniPlugins|to_entries[]| select (.key | contains("clus0-0")).value[0].cniVersion' or 
'.claim.nodes.cniPlugins|to_entries[]| select (.key | contains("clus0-0")).value[0].plugins[].ipam'

The user could use this output to get more details about the differences without overloading the output of this command.

Reply:
I still want to keep the table as a summary of the fields whose value don't match, but I could add the json paths right after it. What do you think?

Reply:
Ok. As discussed in private, this extra improvements will be done in further PRs.